src/badges/views.py

```diff
@@ -106,12 +105,22 @@ def badge_create(request):
 @non_public_only_view
 @staff_member_required
 def badge_copy(request, badge_id):
-    new_badge = get_object_or_404(Badge, pk=badge_id)
-    new_badge.pk = None  # autogen a new primary key (quest_id by default)
-    new_badge.import_id = uuid.uuid4()
-    new_badge.name = "Copy of " + new_badge.name
+    badge = get_object_or_404(Badge, pk=badge_id)
+    
+    form_data_initial = model_to_dict(badge)
+    form_data_initial['tags'] = badge.tags.all()
+
+    # rename copied badge
+    # strips front if len(name + ' - copy') is too long
+    char_limit = BadgeForm._meta.model._meta.get_field('name').max_length
```

Comment:
I think this is all unnecessary.  The badge form will show a super long name, but the form won't submit and the user can change the name if too long.  Not a problem and leaves the full name for the user to decide what to change

Reply:
removed